src/main/java/gov/nasa/jpl/mbee/mdk/mms/sync/manual/ManualSyncRunner.java

```diff
@@ -120,28 +121,41 @@ private File searchForServerElements(Project project, Element element, ProgressS
             nodeIds.add(element.getLocalID());
         }
         File sendData = MMSUtils.createEntityFile(this.getClass(), ContentType.APPLICATION_JSON, nodeIds, MMSUtils.JsonBlobType.SEARCH);
-        HttpRequestBase searchRequest = MMSUtils.prepareEndpointBuilderBasicJsonPostRequest(MMSSearchEndpoint.builder(), project, sendData)
+        //TODO: Return to recursive search by rootElements / ownerID it was crashing MMS4 Elasticsearch
+//        HttpRequestBase searchRequest = MMSUtils.prepareEndpointBuilderBasicJsonPostRequest(MMSSearchEndpoint.builder(), project, sendData)
+//                .addParam(MMSEndpointBuilderConstants.URI_PROJECT_SUFFIX, Converters.getIProjectToIdConverter().apply(project.getPrimaryProject()))
+//                .addParam(MMSEndpointBuilderConstants.URI_REF_SUFFIX, MDUtils.getBranchId(project)).build();
+        HttpRequestBase searchRequest = MMSUtils.prepareEndpointBuilderBasicGet(MMSElementsEndpoint.builder(), project)
                 .addParam(MMSEndpointBuilderConstants.URI_PROJECT_SUFFIX, Converters.getIProjectToIdConverter().apply(project.getPrimaryProject()))
                 .addParam(MMSEndpointBuilderConstants.URI_REF_SUFFIX, MDUtils.getBranchId(project)).build();
+
         // use endpoint to make request
         return MMSUtils.sendMMSRequest(project, searchRequest, progressStatus);
     }
 
-    private static void collectClientElementsRecursively(Project project, Element element, int depth, List<Pair<Element, ObjectNode>> elements) {
+    private static void collectClientElementsRecursively(Project project, Element element, int depth, List<Pair<Element, ObjectNode>> elements, List<String> elementIds) {
         ObjectNode jsonObject = Converters.getElementToJsonConverter().apply(element, project);
         if (jsonObject == null) {
             return;
         }
-        elements.add(new Pair<>(element, jsonObject));
+        String id = element.getLocalID();
+        if (!elementIds.contains(id)) {
+            elements.add(new Pair<>(element, jsonObject));
+            elementIds.add(id);
+        }else {
+            return;
+        }
+
         if (depth-- != 0) {
             for (Element elementChild : element.getOwnedElement()) {
-                collectClientElementsRecursively(project, elementChild, depth, elements);
+                collectClientElementsRecursively(project, elementChild, depth, elements, elementIds);
             }
         }
         if (element.equals(project.getPrimaryModel())) {
+            elementIds.add(Converters.getIProjectToIdConverter().apply(project.getPrimaryProject()));
```

Comment:
What was the intent here? The ID of the project should never match the ID of any of its elements.

Reply:
I was mirroring the logic in the search for server elements above.